### PR TITLE
Refactor header and header dropdown nav

### PR DESCRIPTION
**Pull request by perryharlock**
- The button inserted using JS (inside an anchor) has been removed and replaced with just a button
- The Contributing page has been moved to the dropdown menu
- The dropdown will appear on hover for devices that do not support JS
- The dropdown is otherwise triggered by JS
- Media query variables have been added in preparation for a restyle of the site

### CODE REVIEW

File: _includes/header.html [hunk 1]
```diff
@@ -1,76 +1,80 @@
...
+						<li class="site-nav__item site-nav__item--sub">
+							<a href="/contributing" tabindex="0" class="site-nav__link {% if page.url == '/contributing/' %}site-nav__link--current{% endif %}">
+								Overview
+							</a>
+						</li>
 
-						<li class="site-navigation__item  site-navigation__item--sub-nav  site-navigation__item {% if page.url == '/contributing/code-of-conduct/' %}site-navigation__item--current{% endif %}">
-							<a href="/contributing/code-of-conduct/" tabindex="-1" class="site-navigation__item-link">
+						<li class="site-nav__item site-nav__item--sub">
+							<a href="/contributing/code-of-conduct/" tabindex="0" class="site-nav__link {% if page.url == '/contributing/code-of-conduct/' %}site-nav__link--current{% endif %}">
```
Comment (hollsk) [inline _includes/header.html]: I think the tabindexes can be removed entirely. `a` is already focusable by default, so having `tabindex="0"` doesn't do anything. 

I don't really understand why the `tabindex="-1"` existed on these links in the first place 🤔I assume it must have been something for preventing people accidentally tabbing into them when the menu was closed? (??? or something)

Comment (perryharlock) [reply _includes/header.html]: Yeah possibly....I wasn't sure either. I'll take those out then :-)

File: _includes/header.html [hunk 2]
```diff
@@ -1,76 +1,80 @@
...
 		<span class="site-brand__name">Pa11y</span>
 	</a>
-	<div class="site-navigation" aria-label="Main Navigation">
-		<nav role="navigation">
-			<ul class="site-navigation__menu clearfix" tabindex="0" role="menubar" aria-controls="st1" >
 
-				<li class="site-navigation__item {% if page.url == '/' %}site-navigation__item--current{% endif %}" >
-					<a href="/" class="site-navigation__item-link">
+	<div class="site-nav" aria-label="Main Navigation">
+		<nav role="navigation">
```
Comment (hollsk) [inline _includes/header.html]: the aria label would better on the `<nav>` element (since that's what it's labelling)

File: _includes/header.html [hunk 3]
```diff
@@ -1,76 +1,80 @@
...
+				<li class="site-nav__item">
+					<a href="/" class="site-nav__link {% if page.url == '/' %}site-nav__link--current{% endif %}">
 						Home
 					</a>
 				</li>
 
-				<li class="site-navigation__item {% if page.url contains '/news/' %}site-navigation__item--current{% endif %}" >
-					<a href="/news/" class="site-navigation__item-link">
+				<li class="site-nav__item">
+					<a href="/news/" class="site-nav__link {% if page.url contains '/news/' %}site-nav__link--current{% endif %}">
```
Comment (hollsk) [inline _includes/header.html]: might be fancy to add aria-current to the current menu item too ✨ 

https://tink.uk/using-the-aria-current-attribute/

File: _includes/header.html (hunk 2 above)
Comment (perryharlock) [reply _includes/header.html]: Actually..I'm not even sure that `<div>` is required....

**Pull request merged**